integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateListeners.scala

```diff
@@ -111,6 +113,29 @@ trait CommitHelper {
     }
   }
 
+  def mergeTableStatusContents(uuidTableStatusPath: String,
+      tableStatusPath: String): Boolean = {
+    try {
```

Comment:
Please try to move taking lock inside this method

Reply:
moved lock acquiring locking inside this method